### ISSUE TITLE
Grid doesn't render components if grid is created by e.g. a button click

### ISSUE THREAD

**Issue by KiteDX** [vaadin/vaadin-grid-flow#710]
Vaadin 14.0.0.rc4 (also occurs in previous versions <rc4 and <=13.0.10)

If a grid isn't rendered on the page from the beginning because e.g. you have a button that will only create a grid after the button was pressed, the grid will appear on the page, but it's component columns (does not matter which components you use) are empty while normal text columns will display their content correctly.

In this situation the sequence in which you add the columns to the grid seems to be important:
- if you add the columns BEFORE you add the grid to the layout, everything is ok
- if you add the columns AFTER you add the grid to the layout, the components will not show up

Due to the way our builders work, this feature is crucial for us.

Minimal working example:
Under normal circumstances it will make no difference wether you call createTableAddBeforeAddColumn() or createTableAddAfterAddColumn().

```
@Route("")
public class MainView extends VerticalLayout {
    private static List<String> DATA = Arrays.asList(
            "A", "B", "C"
    );

    public MainView() {
        // works
        createTableAddBeforeAddColumn("Immediate: add(grid) before addColumn(...)");

        // works
        createTableAddAfterAddColumn("Immediate: add(grid) after addColumn(...)");


        Button button = new Button("Show additional tables");
        button.addClickListener((e) -> {
            // does not work
            createTableAddBeforeAddColumn("Delayed: add(grid) before addColumn(...)");
            // works
            createTableAddAfterAddColumn("Delayed: add(grid) after addColumn(...)");
        });
        add(button);
    }

    private void createTableAddBeforeAddColumn(String label) {
        add(new Label(label));

        Grid grid = new Grid();
        grid.setDataProvider(new ListDataProvider(DATA));

        // grid added before addColumn
        add(grid);
        
        grid.addComponentColumn((e) -> new Button("Click me!")).setHeader("Component");
    }

    private void createTableAddAfterAddColumn(String label) {
        add(new Label(label));

        Grid grid = new Grid();
        grid.setDataProvider(new ListDataProvider(DATA));

        grid.addComponentColumn((e) -> new Button("Click me!")).setHeader("Component");

        // grid added after addColumn
        add(grid);
    }
}
```

Result (after clicking the button):
![image](https://user-images.githubusercontent.com/24498956/61044224-6ab4ad80-a3d8-11e9-82d5-ba458a2bc4ea.png)

EDIT:
Removed missing Person-class in DATA-List and replaced it with a normal String (the value isn't used in the example anyway).

**Comment by petrixh** [comment]
Ran into this as well during a migration, still present on RC6.

**Comment by alejandro-du** [comment]
Related discussion: https://vaadin.com/forum/thread/17788530/grid-with-componet-renderers

**Comment by Legioth** [comment]
Should this be closed now when https://github.com/vaadin/flow/pull/6144 has been merged?

**Comment by tmattsso** [comment]
Also just tested with 14.0.2, NumberRenderer is still broken

**Comment by jcgueriaud1** [comment]
I think the fix is not in vaadin 14.0.2 (flow 2.0.8). I tried on my computer (with the fix) and it's working.
Here the test case https://github.com/vaadin/vaadin-grid-flow/pull/730/files#diff-9d5fd6ccbdaa7758d6e32866ea7f1ba3R36

In the test case I'm using NativeButtonRenderer, I also tried locally with NumberRenderer and it's working

And I think this bug is fixed by https://github.com/vaadin/flow/pull/6144 .

**Comment by Legioth** [comment]
The last comment in https://github.com/vaadin/flow/pull/6144 claims that it would be fixed in Flow 2.0.6 and thus also be part of Vaadin 14.0.2. If this is not the case, then we'd have to investigate which part of the chain is broken. Maybe @ujoni would have some clues?

**Comment by ujoni** [comment]
@Legioth My only clue is that I've mislabeled the ticket. It is 12 days old, while `2.0.8` is 17 days old. So it is coming out with `2.0.9` in `14.0.3`, which in turn should come out tomorrow (3 of September, 2019)
